db_schema.sql

```diff
@@ -0,0 +1,112 @@
+DROP SCHEMA public CASCADE;
+CREATE SCHEMA public;
+
+CREATE TABLE Roles (
+	role_id		INT PRIMARY KEY,
+	role_name	TEXT,
+	description	TEXT
+);
+
+CREATE TYPE permission_type as ENUM ();
+CREATE TABLE Permissions (
+	permission_id		INT PRIMARY KEY,
+	permission_title	TEXT,
+	permission_type		permission_type,
+	description			TEXT
+);
+
+CREATE TABLE Role_Permission (
+	role_id			INT REFERENCES Roles(role_id) ON DELETE CASCADE,
+	permission_id	INT	REFERENCES Permissions(permission_id) ON DELETE CASCADE,
+	PRIMARY KEY (role_id, permission_id)
+);
+
+CREATE TABLE Status (
+	status_id		INT PRIMARY KEY,
+	status_name		TEXT,
+	description		TEXT
+);
+
+CREATE TABLE Process (
+	process_id		INT PRIMARY KEY,
+	process_name	TEXT,
+	min_amount		FLOAT,
+	max_amount		FLOAT,
+	status			INT REFERENCES Status(status_id),
+	approvers		TEXT --NOT SURE (I think we dont need this?)
+);
+
+CREATE TABLE Approvers (
+	role_id		INT REFERENCES Roles(role_id) ON DELETE CASCADE,
+	process_id	INT REFERENCES Process(process_id) ON DELETE CASCADE,
+	precedence	TEXT, --NOT SURE
+	PRIMARY KEY (role_id, process_id)
+);
+
+CREATE TABLE Users (
+	user_id			INT PRIMARY KEY,
+	name			TEXT,
+	password		TEXT, -- can store password here?
+	salt			TEXT,
```

Comment:
I think we don't need to store the `salt` in our db. We can follow this implementation for hashing the password: https://medium.com/@jcox250/password-hash-salt-using-golang-b041dc94cb72